### PR TITLE
ROX-18606: Add multi-arch support for s390x and ppc64le using docker buildx

### PR DESCRIPTION
**Pull request by kcrane**
Add multi-arch support for s390x and ppc64le using docker buildx

tested local builds with platforms: amd64, s390x, ppc64le

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,15 +39,34 @@ DEFAULT_IMAGE_REGISTRY := quay.io/stackrox-io
...
+                --build-arg="RPMS_BASE_TAG=stream9" \
+                --build-arg="BASE_IMAGE=ubi9-minimal"
+endif
+
+# Update the arch to arm64 but only for Macs running on Apple Silicon (M1)
+ifeq ($(GOARCH),)
 ifeq ($(shell uname -ms),Darwin arm64)
 	# TODO(ROX-12064) build these images in the CI pipeline
 	# Uncomment the line below to enable native arm64 builder images
 	# BUILD_IMAGE = quay.io/rhacs-eng/sandbox:apollo-ci-stackrox-build-0.3.49-arm64
```
Comment (RTann) [inline Makefile:59]: I think we can get rid of this, finally

File: Makefile [hunk 2]
```diff
@@ -39,15 +39,34 @@ DEFAULT_IMAGE_REGISTRY := quay.io/stackrox-io
...
+DB_DOCKERBUILD_ARGS =
+ifeq ($(GOARCH),s390x)
+        DB_DOCKERBUILD_ARGS = \
+                --build-arg="RPMS_REGISTRY=quay.io" \
+                --build-arg="RPMS_BASE_IMAGE=centos/centos" \
+                --build-arg="RPMS_BASE_TAG=stream9" \
+                --build-arg="BASE_IMAGE=ubi9-minimal"
+endif
+
+# Update the arch to arm64 but only for Macs running on Apple Silicon (M1)
```
Comment (RTann) [inline Makefile:54]: This comment doesn't make sense now that we support multiple architectures (this is from https://github.com/stackrox/stackrox/commit/d1af797460eb38a2b9b7eee4cd34a7a9bcc9301b which was a long time ago)

File: Makefile [hunk 3]
```diff
@@ -39,15 +39,34 @@ DEFAULT_IMAGE_REGISTRY := quay.io/stackrox-io
 BUILD_IMAGE_VERSION=$(shell sed 's/\s*\#.*//' BUILD_IMAGE_VERSION)
 BUILD_IMAGE := $(DEFAULT_IMAGE_REGISTRY)/apollo-ci:$(BUILD_IMAGE_VERSION)
 
+GOINSTALL := GOARCH="" go install
```
Comment (RTann) [inline Makefile:42]: curious what happens without this?

File: Makefile [hunk 4]
```diff
@@ -39,15 +39,34 @@ DEFAULT_IMAGE_REGISTRY := quay.io/stackrox-io
...
+DOCKERBUILD := $(CURDIR)/scripts/docker-build.sh
+
+DB_DOCKERBUILD_ARGS =
+ifeq ($(GOARCH),s390x)
+        DB_DOCKERBUILD_ARGS = \
+                --build-arg="RPMS_REGISTRY=quay.io" \
+                --build-arg="RPMS_BASE_IMAGE=centos/centos" \
+                --build-arg="RPMS_BASE_TAG=stream9" \
+                --build-arg="BASE_IMAGE=ubi9-minimal"
+endif
```
Comment (RTann) [inline Makefile:52]: does this need to be after the GOARCH stuff below?

File: image/db/rhel/Dockerfile.slim [hunk 5]
```diff
@@ -65,6 +81,7 @@ RUN curl -sSLf https://download.postgresql.org/pub/repos/yum/keys/PGDG-RPM-GPG-K
 USER 70:70
 
 ENV ROX_SLIM_MODE="true"
+COPY --from=extracted_bundle /bundle/docker-entrypoint-initdb.d/definitions.sql.gz /docker-entrypoint-initdb.d/
```
Comment (RTann) [inline image/db/rhel/Dockerfile.slim:84]: This is meant to be left out of the `slim` image

File: Makefile (hunk 4 above)
Comment (kcrane) [reply Makefile:52]: good catch... I was always setting GOARCH on make invocation.

File: Makefile [hunk 6]
```diff
@@ -230,24 +245,38 @@ $(CURDIR)/image/db/rhel/bundle.tar.gz:
...
 .PHONY: db-image-slim
 db-image-slim: $(CURDIR)/image/db/rhel/bundle.tar.gz
 	@echo "+ $@"
 	@test -f image/db/dump/definitions.sql.gz || { echo "FATAL: No definitions dump found in image/dump/definitions.sql.gz. Exiting..."; exit 1; }
-	@docker build -t scanner-db-slim:$(TAG) $(IMAGE_BUILD_ARGS) --build-arg POSTGRESQL_ARCH=${ARCH} -f image/db/rhel/Dockerfile.slim image/db/rhel
+	$(DOCKERBUILD) \
+	-t scanner-db-slim:$(TAG) \
+	$(IMAGE_BUILD_ARGS) \
+       $(DB_DOCKERBUILD_ARGS) \
+       -f image/db/rhel/Dockerfile.slim image/db/rhel
```
Comment (RTann) [inline Makefile:279]: nit: spacing

**Comment by RTann** [comment]
CI failures are unrelated

**Review by RTann — APPROVED** [review]
Just the one thing, then LGTM

**Pull request merged**